### DISCUSSION TITLE
Can't use Request parameter with eg Form or other objects built from Request.

### DISCUSSION THREAD

**Discussion by callamd** [fastapi/fastapi#13058] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
https://github.com/callamd/fastapi-issue
```

### Description

If you use Request and say Annotated form as a parameter to a route handler, you get a useless error.

### Operating System

Linux

### FastAPI Version

0.115.6

### Pydantic Version

2.10.3

### Python Version

3.12.7

### Additional Context

https://github.com/callamd/fastapi-issue

**Comment by luzzodev** [comment, marked as answer]
There is an error in your example, you should type the request input but you declaring two different inputs.
```
# Wrong
def post_form(request, Request, form: t.Annotated[forms.StudentForm, Form()]): 

# Corrected
def post_form(request: Request, form: t.Annotated[forms.StudentForm, Form()]):
```

``` python
app_a = FastAPI()
@app_a.post("/form")
def post_form(request: Request, form: t.Annotated[forms.StudentForm, Form()]):
    return {
        "name": form.name,
        "degree": form.degree,
    }
```
